extensions/sql-migration/src/wizard/skuRecommendation/skuRecommendationPage.ts

```diff
@@ -871,446 +339,151 @@ export class SKURecommendationPage extends MigrationWizardPage {
+			return;
+		}
 
-				else if (this.migrationStateModel.performanceCollectionStopped()) {
-					await this._skuDataCollectionStatusIcon.updateProperties({
-						iconPath: IconPathHelper.stop
-					});
-					this._skuDataCollectionStatusText.value = constants.AZURE_RECOMMENDATION_STATUS_STOPPED;
-					this._skuDataCollectionTimerText.value = '';
-
-					await this._skuGetRecommendationContainer.updateCssStyles({ 'display': 'none' });
-					await this._skuDataCollectionStatusContainer.updateCssStyles({ 'display': 'block' });
-					await this._skuStopDataCollectionButton.updateCssStyles({ 'display': 'none' });
-					await this._skuRestartDataCollectionButton.updateCssStyles({ 'display': 'block' });
-					await this._refreshAzureRecommendationButton.updateCssStyles({ 'display': 'none' });
-					await this._skuEditParametersContainer.updateCssStyles({ 'display': 'block' });
-				}
+		// dbReady are thos databases without issues. dbNotReady are those databases with atleast one issue with issueCategory = "Issue".
+		// dbReadyWithWarnings can be found by subtraction dbReady and dbNotReady with total dbCount.
+		const dbReady = this.migrationStateModel._assessmentResults?.databaseAssessments?.filter(db =>
+			!db.issues?.some(issue => issue.appliesToMigrationTargetPlatform === targetType)
+		).length;
+		const dbNotReady = this.migrationStateModel._assessmentResults?.databaseAssessments?.filter(db =>
+			db.issues?.some(issue => (issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === "Issue"))
+		).length;
+		const dbReadyWithWarnings = dbCount - (dbReady + dbNotReady);
+
+		// blockers contain count of all the instance level issues with issueCategory = "Issue".
+		var blockers = this.migrationStateModel._assessmentResults?.issues.filter(issue =>
+			(issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === "Issue")).length;
+		// also blockers includes sum of all the dabatase level issues with issueCategory = "Issue" for each database.
+		blockers += this.migrationStateModel._assessmentResults?.databaseAssessments?.reduce((count, database) =>
+			count + database.issues.filter(issue => (issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === "Issue")).length, 0);
+
+		// warnings contain count all the instance level issues with issueCategory = "Warning".
+		var warnings = this.migrationStateModel._assessmentResults?.issues.filter(issue =>
+			(issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === "Warning")).length;
+		// also warnings includes sum of all the dabatase level issues with issueCategory = "Warning" for each database.
+		warnings += this.migrationStateModel._assessmentResults?.databaseAssessments?.reduce((count, database) =>
+			count + database.issues.filter(issue => (issue.appliesToMigrationTargetPlatform === targetType) && (issue.issueCategory === "Warning")).length, 0);
+
```

Comment:
I recommend creating well named constants for category strings like "Warning"

Reply:
Done.